src/main/java/edu/harvard/iq/dataverse/engine/command/impl/FinalizeDatasetPublicationCommand.java

```diff
@@ -104,6 +104,8 @@ public Dataset execute(CommandContext ctxt) throws CommandException {
 
         // Remove locks
         ctxt.engine().submit(new RemoveLockCommand(getRequest(), theDataset, DatasetLock.Reason.Workflow));
+        ctxt.engine().submit(new RemoveLockCommand(getRequest(), theDataset, DatasetLock.Reason.pidRegister));
```

Comment:
Is there a specific need to attempt to remove the pidRegister lock TWICE? - by the RemoveLockCommand, and then by the direct call to ctxt.datasets().removeDatasetLocks() in the line below? 
Could this be causing some kind of a conflict?
I would leave the latter and remove the former maybe...